pkg/cluster/check/kubernetes.go

```diff
@@ -66,10 +69,7 @@ func K8sAllNodesReportedAssertion(ctx context.Context, cluster ClusterInfo) erro
 		}
 	}
 
-	sort.Strings(expectedNodes)
-	sort.Strings(actualNodes)
-
-	if reflect.DeepEqual(expectedNodes, actualNodes) {
+	if isSubset(allExpectedNodeIPs, actualNodes) {
```

Comment:
this feels like it should be changed, as we match on any expected address of the node, while later we assert that expected IP [0] was actually matched

Reply:
I introduced a function, `assertNodes` which does assertion by comparing expectedNodes<>actualNodes, asserting that they are the same set of nodes (by their internalips) + for each node, asserting that the actual nodes IPs are a **subset** of the expected one. Can you have a look into that?